tests/test_downloader.c

```diff
@@ -55,7 +55,7 @@ START_TEST(test_downloader_single_file)
 
     tmpfn1 = lr_pathconcat(test_globals.tmpdir, "single_file_XXXXXX", NULL);
 
-    mktemp(tmpfn1);
+    fail_if(mkstemp(tmpfn1) == -1);
```

Comment:
* The original code uses `mktemp`. The `mktemp` function generates a unique temporary filename from template.
* The new code uses `mkstemp`. The `mkstemp()` function generates a unique temporary filename from template, **creates and opens the file, and returns an open file descriptor for the file** . The code compares the returned descriptor with -1 and discards it. **File descriptor leak**. On the next line, the code reopens an already open file - the `open` function. 

The problem occurs in many places.

Reply:
I see, thank you.

I have changed it to use the file descriptor from the `mkstemp` call directly and added closing it at the end of tests.
(I have also tried using `_cleanup_fd_close_` which librepo defines but covscans were complaining about it.)